### PR TITLE
[Issue #164] gas and return data helpers for Computation

### PR DESCRIPTION
**Pull request by djrtwo**
### What was wrong?
Issue #164 

### How was it fixed?
- added `should_burn_gas` to Computation and used throughout where applicable
- added `should_erase_return_data` to Computation and used throughout where applicable
- change `VMError` and `Revert` attribute `zeros_return_data`  to `erases_return_data`

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: evm/vm/computation.py [hunk 1]
```diff
@@ -104,6 +104,14 @@ def is_success(self):
     def is_error(self):
         return not self.is_success
 
+    @property
+    def should_burn_gas(self):
+        return self.error and self.error.burns_gas
```
Comment (pipermerriam) [inline evm/vm/computation.py]: Can you adjust this to use the `is_error` property and move `self.error` to be private at `self._error`.  Realizing that as part of this work, `self._error` should be clearly marked private to discourage using it.

**Comment by pipermerriam** [comment]
I suspect you've seen it but: https://travis-ci.org/ethereum/py-evm/jobs/310112334

Some references to `zeros_return_data` still hanging around.

File: evm/vm/computation.py (hunk 1 above)
Comment (djrtwo) [reply evm/vm/computation.py]: on it

File: evm/vm/forks/frontier/__init__.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def _execute_frontier_transaction(vm, transaction):
             # The address of the newly created contract has *somehow* collided
             # with an existing contract address.
             computation = Computation(vm, message)
-            computation.error = ContractCreationCollision(
+            computation._error = ContractCreationCollision(
```
Comment (djrtwo) [inline evm/vm/forks/frontier/__init__.py]: explicit external setting of `_error` occurs in these fork `__init__.py` files. Violates the privateness of `_error`. Thoughts @pipermerriam?

Comment (pipermerriam) [reply evm/vm/forks/frontier/__init__.py]: 🏈   <-- punt on it and mark it with a `# TODO` comment.

**Pull request merged**